### PR TITLE
Check sync status of code with device

### PR DESCRIPTION
**Pull request by microbit-robert**
Basic demo changes to the serial bar to indicate whether the editor code is in sync with the device or not.

### CODE REVIEW

File: src/editor/codemirror/CodeMirror.tsx [hunk 1]
```diff
@@ -92,6 +94,7 @@ const CodeMirror = ({
             undo: undoDepth(view.state),
             redo: redoDepth(view.state),
           });
+          setSyncStatus(SyncStatus.OUT_OF_SYNC);
```
Comment (microbit-matt-hillsdon) [inline src/editor/codemirror/CodeMirror.tsx:97]: Dubious that CM undo/redo is the right place as other file system interactions are possible (e.g. upload/delete file) that are outside the editor. Need similar info from the FS.

But good enough for a demo.

File: src/serial/SerialIndicators.tsx [hunk 2]
```diff
@@ -6,14 +6,33 @@
...
+interface SyncMessage {
+  id: SyncStatus;
+  message: string;
+  color: string;
+}
+
+const syncMessages: SyncMessage[] = [
+  {
+    id: SyncStatus.OUT_OF_SYNC,
+    message: "Running code - out of sync with editor",
```
Comment (microbit-matt-hillsdon) [inline src/serial/SerialIndicators.tsx:26]: Tempted to lose the notion of "Running" entirely.

**Comment by microbit-matt-hillsdon** [comment]
It assumes it's in sync if I unplug and replug the micro:bit after it was previously in sync.

**Comment by microbit-robert** [comment]
Having two icons with the flash button didn't look good and I'm not sure the current 'check' icon looks good either. Unsure about the lighter grey background when the code and device are out of sync.

**Review by microbit-matt-hillsdon — APPROVED** [review]

**Comment by microbit-matt-hillsdon** [comment]
CI failures are during NPM install/ci and seem to be a GC death spiral leading to large amounts of wasted CI time.

This gets stuck for me locally (haven't waited 3+hrs for a timeout):
```
npm i --loglevel verbose -g @microbit-foundation/website-deploy-aws@0.2.0
```

The tail end of the output is all:

```
npm verb attempt to log [
  'ERESOLVE',
  'overriding peer dependency',
  {
    code: 'ERESOLVE',
    edge: { type: 'peer', name: 'react', spec: '^18.0.0', from: [Object] },
    dep: {
      name: 'react',
      version: '17.0.2',
      whileInstalling: [Object],
      location: 'node_modules/react',
      isWorkspace: false,
      dependents: [Array]
    },
    current: {
      name: 'react',
      version: '18.0.0',
      location: 'node_modules/@microbit-foundation/website-deploy-aws/node_modules/react',
      isWorkspace: false,
      dependents: [Array]
    },
    peerConflict: {
      name: 'react',
      version: '18.0.0',
      location: 'node_modules/@microbit-foundation/website-deploy-aws/node_modules/react',
      isWorkspace: false,
      dependents: [Array]
    },
    strictPeerDeps: false,
    force: false,
    isMine: false
  }
]
```

So what changed appears to be the React major release (yesterday).

**Pull request merged**